pinot-core/src/main/java/com/linkedin/pinot/core/query/pruner/AbstractSegmentPruner.java

```diff
@@ -31,7 +32,7 @@
  */
 public abstract class AbstractSegmentPruner implements SegmentPruner {
 
-  public abstract boolean pruneSegment(FilterQueryTree filterQueryTree, Map<String, ColumnMetadata> columnMetadataMap);
+  public abstract boolean pruneSegment(IndexSegment segment, FilterQueryTree filterQueryTree, Map<String, ColumnMetadata> columnMetadataMap);
```

Comment:
It is better to only extract the metadata which is needed. There could be metadata for other pruners which we don't want to extract. Extracting is work which is done once for limited use.